hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/index/IndexResolver.java

```diff
@@ -665,6 +792,10 @@ private static MapIndexScanPhysicalRel createIndexScan(
         // Prepare traits
         RelTraitSet traitSet = OptUtils.toPhysicalConvention(scan.getTraitSet(), distribution);
 
+        // Make a collation trait
+        RelCollation relCollation = buildCollationTrait(scan, index, descending);
```

Comment:
Can we filter out the candidate index early if we realize that it has no collation?

Reply:
Why would we filter it out? Even if there is no collation it might be used by lookup? We don't know on this stage whether an actual sorting exists or not?

Reply:
Added the optimization to the full index scans in https://github.com/hazelcast/hazelcast/pull/18013/commits/91e69a2e72a5296c516d9f240d464ca43225e1f0